src/status_im/contexts/chat/messenger/messages/list/view.cljs

```diff
@@ -102,18 +102,37 @@
      ;; https://github.com/status-im/status-mobile/issues/17426
      [quo/skeleton-list (skeleton-list-props :messages parent-height platform/ios?)]]))
 
+(defn header-height
+  [insets able-to-send-message?]
+  (let [images         (rf/sub [:chats/sending-image])
+        reply          (rf/sub [:chats/reply-message])
+        edit           (rf/sub [:chats/edit-message])
+        link-previews? (or (rf/sub [:chats/link-previews?])
+                           (rf/sub [:chats/status-link-previews?]))]
```

Comment:
is it ok to have subscriptions in the body of non-component functions?

Reply:
hi @clauxx, thank you for reviewing PR.

I think these subscriptions are still bound to the component, as they are directly being called inside the component function itself. We use same approach for [composer](https://github.com/status-im/status-mobile/blob/d7530dfbee64aa6e042b63c402682dbe10b95b43/src/status_im/contexts/chat/messenger/composer/utils.cljs#L191). Also to be sure, I checked re-frisk and didn't saw any subscription leaks.

An example of the leak, where we used subscription in the callback function
https://github.com/status-im/status-mobile/pull/15878/files

@flexsurfer probably knows better about this. Please feel free to correct me.

Reply:
tried calling this function in setTimeout, got this
![Uploading image.png…]()

Reply:
yeah that's fine from re-frame pov, but from the code I would prefer to always have all subscriptions in the component let block